app/src/main/java/org/fossasia/pslab/activity/OscilloscopeActivity.java

```diff
@@ -271,6 +273,18 @@ public void run() {
                             }
                         }
                     }
+
+                    if (!scienceLab.isConnected() || (!isCH1Selected && !isCH2Selected && !isCH3Selected && !isMICSelected)) {
```

Comment:
What if `!scienceLab.isConnected() &&` is used instead of `!scienceLab.isConnected() ||`? :)